packages/tools/src/tools/annotation/UltrasoundAnnotationTool/utils/fanExtraction.ts

```diff
@@ -0,0 +1,248 @@
+  width: number,
+  height: number,
+  fan: FanGeometry,
+  opts: FanExportOptions = {}
+): string {
+  const { center, startAngle, endAngle, innerRadius, outerRadius } = fan;
+  const { strokeStyle = '#0ff', lineWidth = 2, quality = 0.92 } = opts;
+
+  const canvas = document.createElement('canvas');
+  canvas.width = width;
+  canvas.height = height;
+  const ctx = canvas.getContext('2d');
+
+  // Draw the base image
+  const total = width * height;
+  const channels = buffer.length / total;
+  const imgData = ctx.createImageData(width, height);
+  const out = imgData.data;
+
+  for (let i = 0; i < total; i++) {
+    const baseOut = i * 4;
+    if (channels === 1) {
+      const v = buffer[i];
+      out[baseOut] = v;
+      out[baseOut + 1] = v;
+      out[baseOut + 2] = v;
+      out[baseOut + 3] = 255;
+    } else {
+      const baseIn = i * channels;
+      out[baseOut] = buffer[baseIn];
+      out[baseOut + 1] = buffer[baseIn + 1];
+      out[baseOut + 2] = buffer[baseIn + 2];
+      out[baseOut + 3] = channels === 4 ? buffer[baseIn + 3] : 255;
+    }
+  }
+  ctx.putImageData(imgData, 0, 0);
+
+  // Draw fan shape
+  ctx.beginPath();
+
+  // Inner arc (start → end)
+  for (let a = startAngle; a <= endAngle; a += 0.01) {
+    const x = center[0] + innerRadius * Math.cos(a);
+    const y = center[1] + innerRadius * Math.sin(a);
+    if (a === startAngle) {
+      ctx.moveTo(x, y);
+    } else {
+      ctx.lineTo(x, y);
+    }
+  }
+
+  // Outer arc (end → start, reversed)
+  for (let a = endAngle; a >= startAngle; a -= 0.01) {
+    const x = center[0] + outerRadius * Math.cos(a);
+    const y = center[1] + outerRadius * Math.sin(a);
+    ctx.lineTo(x, y);
+  }
+
+  ctx.closePath();
+  ctx.strokeStyle = strokeStyle;
+  ctx.lineWidth = lineWidth;
+  ctx.stroke();
+
+  return canvas.toDataURL('image/jpeg', quality);
+}
+
+export function downloadFanJpeg(
+  imageId: string,
+  contourType: number = 5
+): void {
+  const { contour, simplified, hull, refined, fanGeometry } =
+    calculateFanGeometry(imageId);
+  const { imageBuffer, width, height } = getImageBuffer(imageId) || {};
+  if (!imageBuffer) {
+    return;
+  }
+  let jpegDataUrl;
+  if (contourType === 1) {
+    jpegDataUrl = exportContourJpeg(imageBuffer, width, height, contour);
+  } else if (contourType === 2) {
+    jpegDataUrl = exportContourJpeg(imageBuffer, width, height, simplified);
+  } else if (contourType === 3) {
+    jpegDataUrl = exportContourJpeg(imageBuffer, width, height, hull);
+  } else if (contourType === 4) {
+    jpegDataUrl = exportContourJpeg(imageBuffer, width, height, [
+      refined.P1,
+      refined.P2,
+      refined.P3,
+      refined.P4,
+    ]);
+  } else {
+    jpegDataUrl = exportFanJpeg(imageBuffer, width, height, fanGeometry, {
+      strokeStyle: '#f00', // red outline
+      lineWidth: 3,
+      quality: 0.95,
+    });
+  }
+  saveBinaryData(jpegDataUrl, 'contour.jpg');
+}
+
+export function calculateFanGeometry(imageId: string) {
```

Comment:
Docs and Types for return please, this is public api, also expose that in a utility not just for UltrasoundTool